### ISSUE TITLE
Update `ingress-nginx` to 1.5.1

### ISSUE THREAD

**Issue by snasovich** [rancher/rancher#40224]
Latest RKE1 versions currently added available in KDM are using `ingress-nginx` which is rather old.

Version 1.5.1 (https://github.com/kubernetes/ingress-nginx/releases/tag/controller-v1.5.1) is the latest currently available and we should update to it.

Note that 1.5.1 supports k8s 1.23-1.25 only which aligns nicely with the versions to be introduced by January 2023 k8s patches to Rancher.

At this time it's not in scope to update `ingress-nginx` for k8s 1.22 or earlier.

See https://github.com/rancher/rancher/issues/37925 for a reference of how it was updated to 1.2.1 last time.

**Comment by rishabhmsra** [comment]
**Validated the nginx-ingress-checks**

**Fresh install checks - Pass**
- On rancher v2.7-head(9bb8d56), KDM pointing to dev-v2.7, cluster provisioning using k8s version - 1.25.6-rancher1-1, 1.24.10-rancher1-1 and 1.23.16-rancher1-1 succeeds on all network providers - canal/flannel/weave/calico
- Validated the ingress version and it's updated to `v1.5.1-rancher2`
- Network checks including ingress checks, passed successfully.

**Upgrade checks also passed done as a part of KDM** [checks](https://github.com/rancher/rancher/issues/40210#issuecomment-1415815227), hence closing the issue.

**Comment by snasovich** [comment]
This is waiting on https://github.com/rancher/kontainer-driver-metadata/pull/1089 before it can be tested for v2.7 on https://github.com/rancher/kontainer-driver-metadata/tree/dev-v2.7-for-2.6.11 branch.

**Comment by vivek-infracloud** [comment]
Re-tested the issue on rancher `v2.7.1` with KDM pointing to `dev-v2.7-for-2.6.11` as [commented here](https://github.com/rancher/rancher/issues/40225#issuecomment-1415146212).

nginx ingress is working as expected. ingress and network checks passed.

Hence closing the issue.
